lib/Controller/APIController.php

```diff
@@ -198,9 +191,23 @@ public function delete(int $id): DataResponse {
 			);
 		}
 
-		$this->manager->delete($id);
+		try {
+			$announcement = $this->manager->getAnnouncement($id);
 
-		return new DataResponse();
+			$this->manager->delete($id);
+
+			$user = $this->userSession->getUser();
+			$userId = $user instanceof IUser ? $user->getUID() : '';
+
+			$this->logger->info('Admin ' . $userId . ' deleted announcement: "' . $announcement->getSubject() . '"');
+
+			return new DataResponse();
+		} catch (AnnouncementDoesNotExistException $e) {
+			return new DataResponse(
+				['message' => 'Announcement not found'],
+				Http::STATUS_NOT_FOUND
+			);
+		}
```

Comment:
Let's keep the old APi behaviour of always returning 200 as 404 is not handled by the UI

```suggestion
			$this->logger->info('Admin ' . $userId . ' deleted announcement: "' . $announcement->getSubject() . '"');
		} catch (AnnouncementDoesNotExistException $e) {
		}
		
		return new DataResponse();
```